java/ql/src/experimental/Security/CWE/CWE-755/NFEAndroidDoS.ql

```diff
@@ -0,0 +1,43 @@
+/**
+ * @name Local Android DoS Caused By NumberFormatException
+ * @id java/android/nfe-local-android-dos
+ * @description NumberFormatException thrown but not caught by an Android application that allows external inputs can crash the application, constituting a local Denial of Service (DoS) attack.
+ * @kind path-problem
+ * @tags security
+ *       external/cwe/cwe-755
+ */
+
+import java
+import semmle.code.java.frameworks.android.Intent
+import semmle.code.java.dataflow.FlowSources
+import semmle.code.java.NumberFormatException
+import DataFlow::PathGraph
+
+/**
+ * Taint configuration tracking flow from untrusted inputs to number conversion calls in exported Android compononents.
+ */
+class NFELocalDoSConfiguration extends TaintTracking::Configuration {
+  NFELocalDoSConfiguration() { this = "NFELocalDoSConfiguration" }
+
+  /** Holds if source is a remote flow source */
+  override predicate isSource(DataFlow::Node source) { source instanceof RemoteFlowSource }
+
+  /** Holds if NFE is thrown but not caught */
+  override predicate isSink(DataFlow::Node sink) {
+    exists(Expr e |
+      e.getEnclosingCallable().getDeclaringType().(ExportableAndroidComponent).isExported() and
+      throwsNFE(e) and
+      not exists(TryStmt t |
+        t.getBlock() = e.getEnclosingStmt().getEnclosingStmt*() and
```

Comment:
```suggestion
        t.getBlock() = e.getAnEnclosingStmt() and
```
You could use [`getAnEnclosingStmt()`](https://help.semmle.com/qldoc/java/semmle/code/java/Expr.qll/predicate.Expr$Expr$getAnEnclosingStmt.0.html) here.

Reply:
@Marcono1234 As the code works the same way and <code>getEnclosingStmt*()</code> just allows more levels of nesting. Shall we keep it like it is?

Reply:
<code>get<b>An</b>EnclosingStmt()</code> is equivalent to `getEnclosingStmt().getEnclosingStmt*()`, it is just less verbose (see [source](https://github.com/github/codeql/blob/fc764db8e1498b83c36f2591118c4ffa289f3865/java/ql/src/semmle/code/java/Expr.qll#L67)).

Reply:
Thanks @Marcono1234 for the explanation. I've applied the change.